core/include/vecmem/memory/resources/base_resource.hpp

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include <memory_resource>
+
+namespace vecmem::memory::resources {
+    class base_resource : public std::pmr::memory_resource {
+    public:
+        virtual bool is_host_accessible() const = 0;
+    };
```

Comment:
Note that while we don't need this information in our allocators, I think it still makes sense to flag which resources produce host-accessible memory. Since we'll be using these resources not only in the STL-like allocators.